### DISCUSSION TITLE
localized route support

### DISCUSSION THREAD

**Discussion by kylekirkby** [vercel/next.js#20091] [Ideas]
Localized routes cannot be implemented due to severe routing limitations. 

## Describe the bug

I would like to be able to simply add localized routes to my app. E.g
- /en/posts
  - /en/posts/test-post-id
- /fr/articles
  - /fr/articles/test-post-id
- /de/blogeintrage
  - /de/blogeintrage/test-post-id

I've got the following structure:
![nextjs-illegalstructure](https://user-images.githubusercontent.com/4564433/101897507-8eb68880-3ba2-11eb-807a-ed093e404b50.PNG)

But I keep getting errors complaining about needing a different slug name?

```
You cannot use different slug names for the same dynamic path ('posts' !== 'local').
```

https://github.com/vercel/next.js/issues/9130

## To Reproduce

Try to implement localized routes with the above structure.

## Expected behavior

Support for localized routes / multiple dynamic routes at the same level.

## System information

- OS: Windows 10 / Ubuntu 20.04 WSL2
- Version of Next.js: 10.0.1
- Version of Node.js: v14.10.1

## Additional context

The previous issue seems to have been closed with no actual work around for using localized routes. I may be missing something obvious but from my testing it simply appears to be not possible to implement with the current state of Next.js routing capabilities.

Any thoughts on the matter will be greatly appreciated! 

Thanks in advance!

**Comment by timneutkens** [comment]
Converted to a feature request. You can read what is supported with i18n routing and what is not in the RFC: https://github.com/vercel/next.js/discussions/17078 and the docs: https://nextjs.org/docs/advanced-features/i18n-routing. Localized routes is intentionally not supported initially. There's multiple feature requests for it already.
